hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/atomic/RaftAtomicValueService.java

```diff
@@ -204,4 +204,8 @@ protected final Map<CPGroupId, Object> getSnapshotMap(int partitionId) {
     protected final void clearPartitionReplica(int partitionId) {
         atomicValues.keySet().removeIf(t -> raftService.getCPGroupPartitionId(t.element1) == partitionId);
     }
+
+    public void clearDestroyedValues() {
+        destroyedValues.clear();
```

Comment:
This data structure actually entails keys for all CP groups, not just the CP group the wipe was actioned against. They are `BiTuple<CPGroupId, String>` where the second component is the name of the atomic. I was thinking it may be better to only clear the ones of the CP group that the wipe is actioned against, however it looked like this itself would be more hassle than its worth given that the wipe API currently nukes all CP group destroyed objects.

?

Reply:
@arodionov I agree, second is my preference too. I guess it's just that for some reason the mapping is different for those atomic services vs. the lock service etc for some reason.